### DISCUSSION TITLE
Get the expiration date from the certificates

### DISCUSSION THREAD

**Discussion by rgarrigue** [fastlane/fastlane#18631] [Q&A] [answered]
Hi there

I'm working on a small script to notify me if the certificates come close to their expiration date. 

I already do that with x509 pem: openssl gives me a `notAfter=Apr  9 20:02:43 2022 GMT`, then I sed / date / epoch / compare this and I'm good.

I'm trying to achieve the same on fastlane's certificate, but I'm missing something to convert them to proper pem. Given

```
certs
├── development
│   ├── AAAAAAAAA.cer
│   ├── AAAAAAAAA.p12
└── distribution
    ├── BBBBBBBBBB.cer
    └── BBBBBBBBBB.p12
``` 

I tried 

```
certs/development (master ✔) ᐅ openssl pkcs12 -in AAAAAAAAA.p12 -out test.crt.pem -clcerts -nokeys 
140310107923776:error:0D0680A8:asn1 encoding routines:asn1_check_tlen:wrong tag:../crypto/asn1/tasn_dec.c:1149:
140310107923776:error:0D07803A:asn1 encoding routines:asn1_item_embed_d2i:nested asn1 error:../crypto/asn1/tasn_dec.c:309:Type=PKCS12
```

Since the p12 seems b64 encoded to me, I tried a `base64 -d AAAAAAAAA.p12 > AAAAAAAAA.p12.tmp`  beforehand and openssl -in this .tmp, without success. 

So what am I missing here ? I'ld prefer a pure openssl solution as using fastlane in this monitoring script would requires way more dependencies, but if there's no other choice, will do

Thanks in advance !

**Comment by rgarrigue** [comment]
Looking a bit more at it, 

```
file AAAAAAAAA.p12
AAAAAAAAA.p12: openssl enc'd data with salted password, base64 encoded
```

What is the salt employed ?

**Comment by lucgrabowski** [comment, marked as answer]
Hi, 
if you want to use certificate stored in match repository, you have to decrypt it first, as written in:
http://docs.fastlane.tools/actions/sync_code_signing/#export-distribution-certificate-and-private-key-as-single-p12-file
and then convert to pem file. .p12 files in match repository are only private keys. You can read more in docs linked above.

If you have an unencrypted .cer file, probably can do it in one command:
`openssl x509 -inform der -in <CERT_NAME>.cer -out cert.pem`
